darts/tests/models/forecasting/test_dlinear_nlinear.py

```diff
@@ -253,6 +253,15 @@ def _eval_model(
                 )
                 self.assertLessEqual(e1, 0.40)
                 self.assertLessEqual(e2, 0.34)
+            for model in [DLinearModel, NLinearModel]:
```

Comment:
could you add a check that fitting the model with past covariates and shared_weights=True raises an error?
```suggestion
            # can only fit models with past covariates when shared_weights=False
            for model in [DLinearModel, NLinearModel]:
```